apiserver/admin.go

```diff
@@ -147,6 +156,28 @@ func (a *admin) login(req params.LoginRequest, loginVersion int) (params.LoginRe
 	}, nil
 }
 
+func (a *admin) getAuditRecorder(req params.LoginRequest, authResult *authResult) (*auditlog.Recorder, error) {
+	if !authResult.userLogin || a.srv.auditLogger == nil {
+		return nil, nil
+	}
+	result, err := auditlog.NewRecorder(
+		a.srv.auditLogger,
+		auditlog.ConversationArgs{
+			Who:          req.AuthTag,
+			What:         req.CLIArgs,
+			When:         a.srv.clock.Now(),
+			ModelName:    a.root.model.Name(),
+			ModelUUID:    a.root.model.UUID(),
+			ConnectionID: a.root.connectionID,
+		},
+	)
+	if err != nil {
+		logger.Errorf("couldn't add login to audit log: %T %+v", err, err)
```

Comment:
Do we need the %T here? is this just for debugging? Do we need the logged message here at all, or will the err eventually get logged at the call site?